error.vue

```diff
@@ -82,6 +91,9 @@ const props = defineProps({
     type: Object || undefined,
   },
 });
```

Comment:
Would be kind of nice to use a type for the error; I am assuming this would be an instance of `NuxtError`, which extends `H3Error`:

```ts
declare class H3Error<DataT = unknown> extends Error {
    static __h3_error__: boolean;
    statusCode: number;
    fatal: boolean;
    unhandled: boolean;
    statusMessage?: string;
    data?: DataT;
    cause?: unknown;
    constructor(message: string, opts?: {
        cause?: unknown;
    });
    toJSON(): Pick<H3Error<DataT>, "data" | "statusCode" | "statusMessage" | "message">;
}
```

The `error.default` and `error.type` properties in the current props definition don't seem to be used in the template or JavaScript, but the `error.statusCode` and `error.message` properties _are_ used in the template, but not expected in the props!

I wonder if we can just do:

```
<script setup lang="ts">
  defineProps<{ error: NuxtError }>()
// ...
```